### ISSUE TITLE
What License is this shared under? MIT?

### ISSUE THREAD

**Issue by DkSkydancer** [SqlQuantumLeap/SimpleSqlExec#13]
What license is this released under ? The MIT License (MIT)?

Example:
https://raw.githubusercontent.com/adamralph/stylecop-msbuild/master/license.txt

**Comment by SqlQuantumLeap** [comment]
Hi there. Sorry for the delay in responding but I have been trying to finish something up that is taking longer than I had hoped :frowning:.

And I apologize for not having a clearly stated license. I have been meaning to do that, but wasn't sure which one to go with nor how to apply it both in my code and here in GitHub. I did find a resource &mdash; GNU.org &mdash; that compares various licenses ( http://www.gnu.org/licenses/license-list.html ):

1. https://directory.fsf.org/wiki/License:Expat  (this seems to  identical to the example you linked to)
1. https://directory.fsf.org/wiki/License:X11
1. https://directory.fsf.org/wiki/License:Apache2.0

I guess, for now, given the size of the project, it probably makes sense to go with the Expat / MIT license. I can add that to the source files, and I guess I will poke around the "settings" to see if there is a place to list the license.

**Comment by SqlQuantumLeap** [comment]
@DkSKydancer Sorry, not sure if you got notified of my reply above, but after doing more research I am actually leaning towards going with Apache 2.0 for this project. Does that pose any problems, or do you know of any reason to prefer MIT?

**Comment by bmarinov** [comment]
Hi @SqlQuantumLeap 
I was looking into using this project, but I noticed that the license ist still not being included. Regarding Apache vs MIT: MIT looks a bit friendlier as its not as long. Apache is fine though. Just, if possible, try to license it as soon as possible. Right now I cannot correctly integrate it into our projects or deployment processes :) 

Regards!

**Comment by SqlQuantumLeap** [comment]
@BMarinov Hi there. Sorry for the delay on resolving this issue. Do you know of any operational / practical reason to _not_ go with Apache 2.0? I will start the process of integrating the Apache 2.0 license, but am curious if there is any particular drawback to using it.

**Comment by bmarinov** [comment]
Hi @SqlQuantumLeap, thank you for the quick response!

From what I can tell the biggest practical disadvantage is that the license requires users to state any modifications made to the original product. If a person or company simply uses (or compiles) the CLI tool in its original form, they can freely do so and would only have to package the license together with the executable. But if they fork (and modify) it instead, a list of modifications would need to be included. I am not sure if this is an intended side effect. If it is, then fair enough.

Derived works cannot imply that they are endorsed by the Apache Foundation. Irrelevant. I am not sure if the naming clause prohibits such derived projects from using parts of the original projects’ name.

Otherwise the only disadvantage in my particular case seems to be the somewhat longer time to grasp the Apache 2.0 license and the differences to MIT. 

TL DR: If you want people to clearly state modifications to the original code then Apache, otherwise MIT. 
(Olbigatory "im not a lawyer")

**Comment by SqlQuantumLeap** [comment]
@BMarinov Thanks for that info. And do you know of any practical disadvantages to the MIT license? So far I am still leaning towards the Apache 2.0 license as I do plan on expanding this project next year, and don't want to mess with changing the license.

**Comment by bmarinov** [comment]
@SqlQuantumLeap I dont know of any disadvantages except that the license is maybe a bit too lax. 

If you have specific plans and / or are leaning towards Apache2.0 then sure - its perfectly fine. I dont have an issue with it. Just go ahead and license the library in that manner 👍 .

**Comment by SqlQuantumLeap** [comment]
@BMarinov (and @DkSKydancer ): I have committed the changes necessary to license under the Apache 2.0 License. I believe I did it correctly (added LICENSE.txt file and optional boilerplate notice to code files). Please let me know if I missed anything. If all looks good, I will close this issue.

**Comment by SqlQuantumLeap** [comment]
For future reference (if ever needed), here is some info I found regarding the Apache 2.0 license:

* [MIT, Apache 2 or BSD license: Who is the fairest of them all?](https://snyk.io/blog/mit-apache-bsd-fairest-of-them-all)
* [Why Choose Apache-2.0 License? A Carefully Weighed Decision](https://medium.com/@robmuh/why-choose-apache-2-0-license-4253379b00ce)
* [APPLYING THE LICENSE TO NEW SOFTWARE](https://www.apache.org/dev/apply-license.html#new)
* [APPENDIX: HOW TO APPLY THE APACHE LICENSE TO YOUR WORK](http://apache.org/licenses/LICENSE-2.0.html#apply)
* [Apache License 2.0](https://choosealicense.com/licenses/apache-2.0/)

**Comment by bmarinov** [comment]
Thanks, looks great @SqlQuantumLeap.
